Functions/Public/Get-GitHubGist.ps1

```diff
@@ -16,17 +16,56 @@ function Get-GitHubGist {
       
     .Link
     https://trevorsullivan.net
+    http://dotps1.github.io
     https://developer.github.com/v3/gists/
     #>
-    [CmdletBinding()]
-    param (
+    [CmdletBinding(DefaultParameterSetName = '__AllParameterSets')]
+    [OutputType([System.Object])]
+
+    Param (
+        [Parameter(ParameterSetName = 'Owner')]
+        [String]$Owner = (Get-GitHubAuthenticatedUser).login,
+        [Parameter(ParameterSetName = 'Id')]
+        [String]$Id,
+        [Parameter(ParameterSetName = 'Target')]
+        [ValidateSet('Public', 'Starred')]
+        [String]$Target
     )
     
-    $ApiCall = @{
-        Body = '';
-        RestMethod = '';
-        Method = '';
+    switch ($PSCmdlet.ParameterSetName) {
+        'Owner' {
+            $restMethod = 'users/{0}/gists' -f $Owner
+
+            break
+        }
+        
+        'Id' {
+            $restMethod = 'gists/{0}' -f $Id
+
+            break
+        }
+
+        'Target' {
+            if ($Target -eq 'Public') {
+                $restMethod = 'gists/public'
+            } else {
+                $restMethod = 'gists/starred'
+            }
+
+            break
+        }
+
+        default {
+            $restMethod = 'gists'
+
```

Comment:
Let's get rid of extra, unnecessary white space, like here. There's a few other spots, too. Try to consolidate curly braces, to avoid lots of extra lines of code. Let's try to keep things looking tidy and concise.